RecoBTag/FeatureTools/plugins/DeepDoubleXTagInfoProducer.cc

```diff
@@ -220,48 +224,66 @@ void DeepDoubleXTagInfoProducer::produce(edm::Event& iEvent, const edm::EventSet
                                   trackinfo.getTrackSip2dSig(),
                                   -btagbtvdeep::mindrsvpfcand(svs_unsorted, cand, jet_radius_),
                                   cand->pt() / jet.pt());
+          } else {
+          n_sorted.emplace_back(i, -1, -btagbtvdeep::mindrsvpfcand(svs_unsorted, cand, jet_radius_), cand->pt() / jet.pt()); 
+          n_indexes.push_back(i);
           }
         }
       }
 
-      // sort collections (open the black-box if you please)
+      // sort collections in added order of priority
       std::sort(c_sorted.begin(), c_sorted.end(), btagbtvdeep::SortingClass<std::size_t>::compareByABCInv);
+      std::sort(n_sorted.begin(), n_sorted.end(), btagbtvdeep::SortingClass<std::size_t>::compareByABCInv);
 
-      std::vector<size_t> c_sortedindices;
+      std::vector<size_t> c_sortedindices, n_sortedindices;
 
       // this puts 0 everywhere and the right position in ind
       c_sortedindices = btagbtvdeep::invertSortingVector(c_sorted);
+      n_sortedindices = btagbtvdeep::invertSortingVector(n_sorted);
 
       // set right size to vectors
       features.c_pf_features.clear();
       features.c_pf_features.resize(c_sorted.size());
+      features.n_pf_features.clear();
+      features.n_pf_features.resize(n_sorted.size());
 
       for (unsigned int i = 0; i < daughters.size(); i++) {
         auto const* cand = daughters.at(i);
         if (cand) {
           // candidates under 950MeV are not considered
           // might change if we use also white-listing
-          if (cand->pt() < 0.95)
+          if (cand->pt() < min_candidate_pt_)
             continue;
+          auto packed_cand = dynamic_cast<const pat::PackedCandidate*>(cand);
+          auto reco_cand = dynamic_cast<const reco::PFCandidate*>(cand);
 
+          // need some edm::Ptr or edm::Ref if reco candidates
+          reco::PFCandidatePtr reco_ptr;
+          if (pf_jet) {
+            reco_ptr = pf_jet->getPFConstituent(i);
+          } else if (pat_jet && reco_cand) {
+            reco_ptr = pat_jet->getPFConstituent(i);
+          }
           // get PUPPI weight from value map
           float puppiw = 1.0;  // fallback value
-
-          float drminpfcandsv = btagbtvdeep::mindrsvpfcand(svs_unsorted, cand, jet_radius_);
-
+          // if (reco_cand && use_puppi_value_map_) {
```

Comment:
Commented out code can be remove if not needed (otherwise add one more comment line which specifies why it is needed commented out)